gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static org.locationtech.spatial4j.context.SpatialContext.GEO;
+
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.List;
+import org.locationtech.spatial4j.distance.DistanceCalculator;
+import org.locationtech.spatial4j.distance.DistanceUtils;
+import org.locationtech.spatial4j.shape.Point;
+import org.locationtech.spatial4j.shape.ShapeFactory;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.FastTravelBetweenStopsNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+
+/**
+ * Validates for each trip that the travel speed between each stop is not too fast.
+ *
+ * <p>Generated notice: {@link FastTravelBetweenStopsNotice}.
+ */
+@GtfsValidator
+public class TripTravelSpeedValidator extends FileValidator {
+  private final static float METER_PER_SECOND_TO_KMH_CONVERSION_FACTOR = 3.6f;
+  private final static int FAST_SPEED_METERS_PER_SECOND = 42; // approx 150 km/h
+  private final static int KILOMETER_TO_METER_CONVERSION_FACTOR = 1000;
+  private final static int SECONDS_IN_DAY = 24 * 60 * 60;
```

Comment:
If only it was so simple...

https://www.youtube.com/watch/-5wpm-gesOY

Feel free to resolve this issue as a TODO.

Reply:
Ah interesting. I'm not sure what the best course of action would be considering that the GtfsTime type only stores the hours/minutes/seconds, there is no day/date attached to the time. Since the only information stored is the secondsSinceMidnight, do you have any suggestions or have you used any good libraries before? 

Will add this as a TODO now though.